src/meta/src/hummock/hummock_manager.rs

```diff
@@ -1140,14 +1145,25 @@ where
     }
 
     pub async fn proceed_version_checkpoint(&self) -> risingwave_common::error::Result<()> {
-        let versioning_guard = self.versioning.read().await;
+        let mut versioning_guard = self.versioning.write().await;
         let new_checkpoint = versioning_guard
             .hummock_versions
             .first_key_value()
             .unwrap()
             .1
             .clone();
         new_checkpoint.insert(self.env.meta_store()).await?;
+
+        versioning_guard.checkpoint_version_id = new_checkpoint.id;
+        loop {
+            let fst = versioning_guard.hummock_version_deltas.first_key_value();
+            if fst.is_some() && *fst.unwrap().0 <= versioning_guard.checkpoint_version_id {
+                HummockVersionDelta::delete(self.env.meta_store(), fst.unwrap().0).await?;
```

Comment:
> We should wrap the delete in one transaction instead of deleteing one entry at a time. We can also leverage `split_off` in BTreeMap to truncate version deltas instead of doing a iteration.

fixed